src/ophys_etl/transforms/event_detection.py

```diff
@@ -0,0 +1,557 @@
+    ncpu: int
+        number or workers for the multiprocessing pool
+
+    Returns
+    -------
+    events: np.ndarray
+        event magnitudes. same shape as traces. zero where no event detected.
+    lambdas: np.ndarray
+        length ntrace optimized regularization values.
+
+    """
+    args = [(trace, sigma, gamma)
+            for trace, sigma in zip(traces, noise_estimates)]
+    func = fast_lzero_regularization_search_bracket
+
+    # pytest-cov does not play nice with Pool context manager so explicit
+    # close/join to get code coverage report in subprocess calls
+    pool = multiprocessing.Pool(ncpu)
+    results = pool.starmap(func, args)
+    pool.close()
+    pool.join()
+
+    events, lambdas = zip(*results)
+    events = np.array(events)
+    lambdas = np.array(lambdas)
+    return events, lambdas
+
+
+def trace_resample(traces: np.ndarray, input_rate: float,
+                   target_rate: float = 30.9) -> Tuple[np.ndarray, float]:
+    """determines a rounded upsample factor and upsamples the traces,
+    if applicable.
+
+    Parameters
+    ----------
+    traces: np.ndarray
+        ncell x nframe float
+    input_rate: float
+        sampling rate of traces [Hz]
+    target_rate: float
+        desired resampled rate [Hz]
+
+    Returns
+    -------
+    resampled_traces: np.ndarray
+        ncell x n_upsampled_frames float
+    upsample_factor: int
+        factor used to upsample
+
+    Raises
+    ------
+    NotImplementedError
+        for any target rate != 30.9Hz. Evaluation of results is needed
+        to generalize FastLZero to other target rates.
+
+    """
+    upsample_factor = np.round(target_rate / input_rate)
+    if upsample_factor == 1.0:
+        resampled_traces = traces
+    else:
+        resampled_traces = resample_poly(traces, upsample_factor, 1, axis=1)
+    return resampled_traces, upsample_factor
+
+
+class EventDetection(argschema.ArgSchemaParser):
+    default_schema = EventDetectionInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+        self.logger.setLevel(self.args.pop('log_level'))
+        ophys_etl_commit_sha = os.environ.get("OPHYS_ETL_COMMIT_SHA",
+                                              "local build")
+        self.logger.info(f"OPHYS_ETL_COMMIT_SHA: {ophys_etl_commit_sha}")
+
+        # read in the input file
+        with h5py.File(self.args['ophysdfftracefile'], 'r') as f:
+            roi_names = f['roi_names'][()].astype('int')
+            valid_roi_indices = np.argwhere(
+                    np.isin(
+                        roi_names,
+                        self.args['valid_roi_ids'])).flatten()
+            dff = f['data'][valid_roi_indices]
+        valid_roi_names = roi_names[valid_roi_indices]
+
+        # set rates the same, i.e. never upsample
+        self.logger.warning("hard-coded not to upsample")
+        upsampled_dff, upsample_factor = trace_resample(
```

Comment:
Should there be an `if target_rate==input_rate` block that circumvents any call to `trace_resample`? Or does `trace_resample` still do something, even if the two rates are the same?

Reply:
In this case, `trace_resample` just returns the input. I am hoping the outcome of the follow-up on the data side of things will lead to a complete elimination of upsampling/downsampling, in which case, I'll clean this up as well.